From: Mark McLoughlin <markmc@redhat.com>
Date: Tue, 21 Oct 2008 09:21:13 +0100
Subject: [net] tun: fix printk warning
Message-id: 1224577273.708.8.camel@blaa
O-Subject: tun: fix printk(size_t) warning
Bugzilla: 468536
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/459719

Fix warning:

  drivers/net/tun.c: In function ‘tun_get_user’:
  drivers/net/tun.c:274: warning: format ‘%u’ expects type ‘unsigned int’, but argument 4 has type ‘size_t’

Introduced in 2.6.28-111.el5 by:

  9be7e03 [net] tun: add IFF_VNET_HDR, TUNGETFEATURES, TUNGETIFF

Signed-off-by: Mark McLoughlin <markmc@redhat.com>

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 671bb39..0f5a11d 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -270,7 +270,7 @@ static __inline__ ssize_t tun_get_user(struct tun_struct *tun, struct iovec *iv,
 		if (gso.csum_start + gso.csum_offset > len - 2) {
 			if (net_ratelimit())
 				printk(KERN_WARNING
-				       "bad partial csum: csum=%u/%u len=%u\n",
+				       "bad partial csum: csum=%u/%u len=%zu\n",
 				       gso.csum_start, gso.csum_offset, len);
 			tun->stats.rx_dropped++;
 			kfree_skb(skb);
